plugins/main/windows/l2bridge/l2bridge_windows.go

```diff
@@ -100,14 +99,14 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 		// NAT based on the the configured cluster network
 		if n.IPMasq {
-			n.ApplyOutboundNatPolicy(n.clusterNetworkPrefix.String())
+			n.ApplyOutboundNatPolicy(n.ClusterNetworkPrefix)
 		}
 
 		hnsEndpoint := &hcsshim.HNSEndpoint{
 			Name:           epName,
 			VirtualNetwork: hnsNetwork.Id,
-			DNSServerList:  strings.Join(result.DNS.Nameservers, ","),
-			DNSSuffix:      result.DNS.Domain,
+			DNSServerList:  strings.Join(n.DNS.Nameservers, ","),
+			DNSSuffix:      n.DNS.Domain,
```

Comment:
@rakelkar , I am not sure if this is the proper way, but result variable is always empty when returned from IPAM plugin. Here are how r and result look like:
```
2018/03/05 01:54:01 Data: &types020.Result{CNIVersion:"0.2.0", IP4:(*types020.IPConfig)(0xc0421a22a0), IP6:(*types020.IPConfig)(nil), DNS:types.DNS{Nameservers:[]string(nil), Domain:"", Search:[]string(nil), Options:[]string(nil)}}
2018/03/05 01:54:01 Data: &current.Result{CNIVersion:"0.3.1", Interfaces:[]*current.Interface(nil), IPs:[]*current.IPConfig{(*current.IPConfig)(0xc0421a23c0)}, Routes:[]*types.Route{}, DNS:types.DNS{Nameservers:[]string(nil), Domain:"", Search:[]string(nil), Options:[]string(nil)}}
```

after passing
```
{"IPMasq":true,"clusterNetworkPrefix":"10.200.0.0/16","cniVersion":"0.2.0","dns":{"Domain":"default.svc.cluster.local","Nameservers":["10.201.0.10"]},"ipam":{"subnet":"10.200.9.0/24","type":"host-local"},"name":"cbr0","type":"win-l2bridge"}
```

But netConfig looks okay and using n.DNS.Nameservers and n.DNS.Domain gives proper result, the container interface is configured with DNS as expected:
```
Ethernet adapter vEthernet (d54bd7bb5e200d550d7b0ba1ac70db5340b0806a945790279e86989ee4819571_cbr0):

   Connection-specific DNS Suffix  . : default.svc.cluster.local
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter #5
   Physical Address. . . . . . . . . : 00-15-5D-D5-0A-C1
   DHCP Enabled. . . . . . . . . . . : No
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : fe80::418a:ae59:763b:50cc%29(Preferred)
   IPv4 Address. . . . . . . . . . . : 10.200.9.72(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 10.200.9.2
   DNS Servers . . . . . . . . . . . : 10.201.0.10
   NetBIOS over Tcpip. . . . . . . . : Disabled
```

Reply:
Well it is almost surely something wrong with host-local IPAM, using n.DNS is just a workaround then